### DISCUSSION TITLE
No idea how to build it - send help!

### DISCUSSION THREAD

**Discussion by benny2581** [sofa-framework/sofa#3715] [Build / Config / Environment] [answered]
Hey,
I am pretty new to sofa and all of coding, programming and basically computer stuff. I have no idea what to do to build sofa on my computer. I downloaded and installed all the programms they recommended on the website right here https://www.sofa-framework.org/community/doc/getting-started/build/windows/
But I honestly don't know what I'm supposed to do next. How to use CMake to add all that stuff? I constantly get errors saying something is invalid.
I looked at the video where a guy shows it all but the main problem appears at the very beginning - he has some folders in his path called src or sofabuild32 but on my device they are nowhere to be found so this is where I get confused. 

If anyone could write a step-by-step what to do, or even let me talk to them about how to end my computer suffering, I would be very grateful.

Have a wonderful day, 
B.

**Comment by alxbilger** [comment]
Hi,

Binaries are the easiest way to start if you are a beginner. Is there a specific reason you need to build SOFA from sources?

If you missed them, they are available on the download page: https://www.sofa-framework.org/download/

If you really need to build SOFA from sources, please copy/paste the CMake error you get. Note that the video you mention is probably quite old, and some things may have changed since.

**Comment by benny2581** [comment, marked as answer]
Maybe that’s a stupid question, but what exactly are binaries? I downloaded
sofa from their website and I even got to the point where I could open that
default simulation with a snake, but I thought I needed all of the sources
to do anything there. Meaning, I don’t know how to open any window where I
could code my own simulation.

**Reply by alxbilger** [reply]
Binaries are the resulting files from a compilation. It's the program that you can execute on your system. If you are able to run it and simulate the snake, you don't need to build the program from sources.
The program that you run (it's called `runSofa`) is able to load simulation files written in various formats, including XML or Python. For example, the snake simulation is written in XML and is available in the example folder which is shipped with the program. If you want to write your own simulation, you need to write a scene in the format of your choice (with a text/code editor), and then open it with `runSofa`.

Links:
- https://www.sofa-framework.org/community/doc/using-sofa/runsofa/
- https://www.sofa-framework.org/community/doc/using-sofa/create-your-scene-in-xml/
- https://sofapython3.readthedocs.io/en/latest/index.html

**Comment by benny2581** [comment]
Wow I wish they put such a great explanation for digital newbies on the
website.
Thank you, you are a life saver!!!
